compiler/qsc_project/src/project.rs

```diff
@@ -231,8 +234,64 @@ pub trait FileSystemAsync {
         Ok(files)
     }
 
+    async fn collect_sources_from_files_field(
+        &self,
+        project_path: &Path,
+        manifest: &Manifest,
+    ) -> ProjectResult<Vec<PathBuf>> {
+        let mut v = vec![];
+        for file in &manifest.files {
+            v.push(
+                self.resolve_path(project_path, Path::new(&file))
+                    .await
+                    .map_err(|e| Error::FileSystem {
+                        about_path: project_path.to_string_lossy().to_string(),
+                        error: e.to_string(),
+                    })?,
+            );
+        }
+        Ok(v)
+    }
+
+    /// Compares the list of files in the `src` directory with the list of files
+    /// in the `files` field, and adds errors if any are missing.
+    fn validate_files_list(
+        project_path: &Path,
+        qs_files: &mut Vec<PathBuf>,
+        listed_files: &mut Vec<PathBuf>,
+        errors: &mut Vec<Error>,
+    ) {
+        qs_files.sort();
+        listed_files.sort();
+
+        let mut listed_files = listed_files.iter().peekable();
+
+        for item in qs_files.iter() {
+            while let Some(&next) = listed_files.peek() {
+                if next < item {
```

Comment:
I'm not understand this just at a glace. What does it mean to compare to PathBufs with a `<`?

Reply:
Probably lexical comparison, but the implementation doesn't really matter here. The point is that I want to compare these to vectors item by item. And sorting them and iterating over both of them in the same order, was a practical way to do it. This `if` just skips over the items in `listed_files` iterator until it gets to one that it can compare with the current item in `qs_files` (next line).